### PR TITLE
Squads 3!

### PR DESCRIPTION
**Pull request by covertcorvid**
## About The Pull Request
Overhauled the squads system
- Re-added squad duties
  - Your squad can be assigned to damage control, medical, munitions, CIC ops, security fireteam, or combat air patrol.
  - Shortly after the start of the shift, the game will attempt to automatically assign squads to CIC Ops, Munitions, or Damage Control if bridge staff, munitions techs, or engineers are not present
  - Declaring a medical, security, or engineering emergency with a request console now alerts the relevant squad in addition to the department
  - The request console of a department with a related squad type can be used to page those squads (engineering to damage control, munitions to munitions support, etc)
- Made squad pagers more pager-like
  - Pager messages are no longer spoken aloud and do not make a radio crackle noise
  - The last message received can be seen by using your pager in hand
  - Most squad members can only receive messages with their pager. Messages can be sent by:
    - A squad management console or tablet program
    - The squad's leader
    - Request consoles, as noted above
    - The AI
  - Messages to squad pagers are limited to 50 characters
  - Fixed punctuation getting converted to number codes
  - Squadchat no longer calls everyone a midshipman
  - Squadchat no longer double-messages you
- Default squad leadership tries to put the highest-ranking person in charge
- Squad members no longer have in-squad roles of marine, medic, engineer, or sergeant. You're either the leader or you aren't.

## Why It's Good For The Game
Multi-squad boarding missions aren't happening very often. Being able to assign squads to specific tasks was used more often. This also lets people more easily tell squads where they're needed.

## Changelog
:cl:
refactor: Reorganized all of the squad code
add: Re-added squad duty assignments
add: Squads with certain duties will now receive department console emergency alerts
add: Departments with related squads can use their request console to page those squads
del: Removed in-squad roles (engineer, medic, marine)
del: Removed squad rank overrides
tweak: Squad tries to pick the highest ranking individual as leader by default
tweak: Pagers are not verbal, limited to short messages, and most are receive-only.
tweak: You can see the last message received by a pager by using it in hand
tweak: Visual improvements to the squad manager
fix: Squad management UI now updates when changes are made
fix: EMTs and heads should not be put in squads anymore
fix: Punctuation now works in pager messages
fix: Pager messages are no longer doubled
/:cl:

### CODE REVIEW

**Comment by Karmic-Skink** [comment]
https://youtu.be/D-fAqJei1nU

Suggestion: Pager gets louder if not answered within 5s, take 1 hearing damage per second past the initial 5s.

Realism.

**Comment by covertcorvid** [comment]
well that didn't work right

**Comment by Absolutely-Steph** [comment]
some small problems, all related to the character limit
image

You'd need to shorten the announcements to something more tactical sounding, and make sure you can't attempt to send a message over the limit, since right now i could type "the quick brown fox jumped over the lazy dog or so|mething" and it'd send everything until hitting 50 characters and then it just cuts off the message. Leading to me having to use a symbol counter website to make sure my messages aren't getting cut off.

**Comment by Kenionatus** [comment]
Checking this out locally, it seems like the medical team doesn't have access to cloning or chemistry. Is this intended? Seems at odds with how damage control has access to the engine room.

Another thing I noticed is that the security squad is called Security Fireteam. Fireteam is the military term for a subdivision inside a squad. To quote Wikipedia:

> "In US usage, a squad consists of eight to fourteen [soldiers](https://en.wikipedia.org/wiki/Soldier),[[6]](https://en.wikipedia.org/wiki/Squad#cite_note-6) and may be further subdivided into fireteams."

**Comment by Bokkiewokkie** [comment]
After using the squads the main thing I still dislike is that returning the squad kits doesn't seem to work. When testing I picked a squad engineer kit, took out the things inside, and it just kept saying I still needed to return 6 items (maybe let us hand in the items one at a time?).
There's also the problem of accidentally folding up the squad boxes and being unable to make new ones to hand in.

**Comment by covertcorvid** [comment]
> After using the squads the main thing I still dislike is that returning the squad kits doesn't seem to work. When testing I picked a squad engineer kit, took out the things inside, and it just kept saying I still needed to return 6 items (maybe let us hand in the items one at a time?). There's also the problem of accidentally folding up the squad boxes and being unable to make new ones to hand in.

lmao so apparently it just checks things in the top level of the backpack. will work on that

File: code/game/objects/items/stacks/sheets/sheet_types.dm [hunk 1]
```diff
@@ -460,6 +460,9 @@ GLOBAL_LIST_INIT(cardboard_recipes, list (														\
 	new/datum/stack_recipe("pizza box", /obj/item/pizzabox),									\
 	new/datum/stack_recipe("folder", /obj/item/folder),											\
 	null,																						\
+	// NSV13 - added squad kit box
+	new /datum/stack_recipe("squad kit box", /obj/item/storage/box/squad_kit),					\
+	null,																						\
```
Comment (Bokkiewokkie) [inline code/game/objects/items/stacks/sheets/sheet_types.dm:465]: Should probably be under "fancy boxes"

File: nsv13/code/modules/squads/squad_vendor.dm [hunk 2]
```diff
@@ -63,71 +71,110 @@
...
-			var/loaned_amount = 0 //How much gear have we loaned them?
-			for(var/path in S.items)
+			var/list/must_return = loadout.items.Copy()
+			//If they don't have weapons clearance, don't expect them to return guns that they don't get vended.
+			if(!H.squad.weapons_clearance)
+				for(var/banned in requires_weapons_clearance)
+					for(var/path in must_return)
+						if(ispath(path, banned))
+							must_return -= path
+			for(var/path in must_return)
```
Comment (Bokkiewokkie) [inline nsv13/code/modules/squads/squad_vendor.dm:92]: Can you make sure to exclude consumable items from this as well? It seems unfair to make them pay thousands of credits for using the consumables we gave them

Comment (covertcorvid) [reply nsv13/code/modules/squads/squad_vendor.dm:92]: It doesn't do any kind of recursive check to make them return consumables - all they have to do is return the box they came in

**Review by Bokkiewokkie — APPROVED** [review]

**Comment by Kenionatus** [comment]
The permissions for the squad management PDA program seem to (still) be scuffed. Squad management program does not appear in Software Download Tool when wearing Captain ID:
image
Version: current master branch at 02ff0c5cc07838a80dc7fde2dc251a3192ee948e

**Pull request merged**